### PR TITLE
Fetching data from backend

### PR DESCRIPTION
**Pull request by ataddesse**
- Imports HttpClient Module
- Generates two service typescript files each for lead and campaign

### CODE REVIEW

File: frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -17,6 +20,10 @@ import { CampaignsComponent } from './campaigns/campaigns.component';
 import { GuideComponent } from './guide/guide.component';
 import { AnalyticsComponent } from './analytics/analytics.component';
 import { IntroComponent } from './intro/intro.component';
+import {MatTableModule} from '@angular/material/table';
```
Comment (r0ddy) [inline frontend/src/app/app.module.ts:23]: Add some spacing before and after `MatTableModule`.

File: frontend/src/app/app.module.ts [hunk 2]
```diff
@@ -17,6 +20,10 @@ import { CampaignsComponent } from './campaigns/campaigns.component';
...
+
+//fixing 404 problem here by using # pr
+import { HashLocationStrategy, LocationStrategy  } from '@angular/common';
```
Comment (r0ddy) [inline frontend/src/app/app.module.ts:26]: Yeah, I was looking into this as well. Seems like this should fix it. We will also need to import `Location` to [normalize](https://angular.io/api/common/HashLocationStrategy) any paths.

File: frontend/src/app/leads/leads.component.ts [hunk 3]
```diff
@@ -1,15 +1,29 @@
 import { Component, OnInit } from '@angular/core';
+import { LeadService } from '../shared/lead.service';
+import { Lead } from './model/lead.model';
+import {MatTableDataSource} from '@angular/material/table';
```
Comment (r0ddy) [inline frontend/src/app/leads/leads.component.ts:4]: Sam nit as before.

File: frontend/src/app/leads/leads.component.ts [hunk 4]
```diff
@@ -1,15 +1,29 @@
...
 export class LeadsComponent implements OnInit {
+   leads : Lead[];
 
-  constructor() { }
+    displayedColumns = ['lead_id','name', 'phone_number', 'campaign_id', 'date'];
+    dataSource: MatTableDataSource<Lead>;
+
+   constructor(private leadService : LeadService) { }
+
+     ngOnInit() : void {
```
Comment (r0ddy) [inline frontend/src/app/leads/leads.component.ts:19]: Fix indentation.

File: frontend/src/app/leads/leads.component.ts [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
+        this.getAllLeads();
+      }
+
+      getAllLeads(): void {
```
Comment (r0ddy) [inline frontend/src/app/leads/leads.component.ts:23]: Fix indentation.

File: frontend/src/app/leads/model/lead.model.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import{ ColumnData } from './column-data.model';
```
Comment (r0ddy) [inline frontend/src/app/leads/model/lead.model.ts:1]: Space after import.

File: frontend/src/app/leads/model/lead.model.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export interface Lead {
+   date:Date;
```
Comment (r0ddy) [inline frontend/src/app/leads/model/lead.model.ts:4]: Add a space after colon.

File: frontend/src/app/shared/lead.service.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import {HttpClient, HttpHeaders } from '@angular/common/http';
+import { Lead } from '../leads/model/lead.model';
+import { Observable, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
+
+
+@Injectable()
+export class LeadService {
+private url = 'http://localhost:8080/api/webhook';
```
Comment (r0ddy) [inline frontend/src/app/shared/lead.service.ts:10]: No need to prepend base url.

File: frontend/src/app/shared/lead.service.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+ httpOptions = {
+
```
Comment (r0ddy) [inline frontend/src/app/shared/lead.service.ts:12]: Delete this unused object.

File: frontend/src/app/shared/lead.service.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+  };
+    constructor(
```
Comment (r0ddy) [inline frontend/src/app/shared/lead.service.ts:16]: Fix indentation here.

File: frontend/src/app/app.module.ts (hunk 1 above)
Comment (ataddesse) [reply frontend/src/app/app.module.ts:23]: Sure

File: frontend/src/app/app.module.ts (hunk 2 above)
Comment (ataddesse) [reply frontend/src/app/app.module.ts:27]: Alright, I think we will be going with your fix for this line when merging.

File: frontend/src/app/leads/leads.component.ts (hunk 3 above)
Comment (ataddesse) [reply frontend/src/app/leads/leads.component.ts:4]: Alright

File: frontend/src/app/shared/lead.service.ts (hunk 8 above)
Comment (ataddesse) [reply frontend/src/app/shared/lead.service.ts:10]: That causing problem for me let me see if there is no any error with just  /api/webhook

**Pull request merged**